test/adapter.jl

```diff
@@ -0,0 +1,16 @@
+@testset "iteration" begin
+    l = GAP.evalstr("[1, 2, 3]")
+    lj = collect(l)
+    @test lj isa Vector{Any}
+    @test lj == [1, 2, 3]
+    lj = collect(Int, l)
+    @test lj isa Vector{Int}
+    @test lj == [1, 2, 3]
+
+    s = GAP.Globals.SymmetricGroup(3)
+    xs = []
+    for x in s
+        push!(xs, x)
+    end
+    @test length(xs) == 6
```

Comment:
So here, we can't do `collect(s)` because `collect` tries to get the length of `s`, which doesn't work. The correct lenght of `s` in this context would be to use `Globals.Order(s)` I believe. So should we make `length` return `Order` here, and similarly for other iterable GAP objects, or simply defining `Base.IteratorSize(x::GapObj) = Base.SizeUnknown()` ?

Reply:
From GAP's point of view, `Globals.Size` is the function that returns the number of elements in the collection.

(And from the viewpoint of what iterators are used for in GAP, if the code would be an example how to proceed then I would not collect the elements in a list but just increase a counter for each element: The main idea in GAP is to avoid creating a full elements list whenever it is likely that either one can stop after one has found a suitable element or at least one creates only one element at a time, which saves space.)

Reply:
> From GAP's point of view, Globals.Size is the function that returns the number of elements in the collection.

Ah thank you! 

Concerning your comment inside parentheses, I think I understand the content, but maybe not the implications for this PR... are you referring to my test code above where I collect the iterators's elements in a list? Usually it's not necessary to "collect" elements of an iterator in a list, but inevitably we collect them anyway once in a while, if only at the REPL to "see" the content of the iterator, for small collections, so supporting `collect` would be good.